client/components/Dashboard.tsx

```diff
@@ -31,47 +29,243 @@ const GET_PROJECT = gql`
+            });
+        } else if (arg === 'complexity') {
+            setStyle({
+                ...style,
+                time: false,
+                depth: false,
+                complexity: true,
+            });
+        }
+    };
+    // graphql calling
     const { data: dataR, loading: loadingR } = useQuery(GET_USER_DATA, {
         variables: {
-            userId: '6286978e12716d47e6884194',
+            userId: '62ba5b743f40d18829e018a1',
         },
     });
     const { data, loading } = useQuery(GET_PROJECT, {
         variables: {
-            projectId: '628e864e76cdbbec53f36010',
+            projectId: '62ba5bd43f40d18829e018c4',
         },
     });
 
-    const [projects, setProjects] = useState<Projects['projects']>();
+    // const [projects, setProjects] = useState<Projects['projects']>();   // commented out due to still not knowing how you guys want our project selection to look like. Waiting for instructions
     const [selectedProject, selectProject] = useState<SelectedProject['project']>();
+    const [queries, selectQuerries] = useState<ProjectQuery[]>();
 
-    const test = (pr: any): void => {
-        selectProject(pr);
-    };
+    // const test = (pr: any): void => { // this is poorly named fuction that selects the project
+    //     selectProject(pr);
+    // };
 
-    useEffect(() => {
-        if (!loadingR && dataR) {
-            setProjects(dataR.user.projects);
+    // sorting functions
+
+    const sortByTime = (): any => {
+        if (order === true) {
+            setOrder(false);
+            // setArrow();
+            setArrow({
+                ...arrow,
+                time: '↑',
+                depth: '',
+                complexity: '',
+            });
+        } else {
+            setOrder(true);
+            setArrow({
+                ...arrow,
+                time: '↓',
+                depth: '',
+                complexity: '',
+            });
+        }
+        if (order) {
+            const newArr = [];
+            if (selectedProject) {
+                // eslint-disable-next-line no-restricted-syntax, guard-for-in
+                for (const key in selectedProject?.queries) {
+                    newArr.push(selectedProject?.queries[key]);
+                }
+            }
+            const dataToSort = [...newArr];
+            dataToSort.sort((a, b) => a.timestamp - b.timestamp);
+            selectQuerries(dataToSort);
+        } else {
+            const newArr = [];
+            if (selectedProject) {
+                // eslint-disable-next-line no-restricted-syntax, guard-for-in
+                for (const key in selectedProject?.queries) {
+                    newArr.push(selectedProject?.queries[key]);
+                }
+            }
+            const dataToSort = [...newArr];
+            dataToSort.sort((a, b) => a.timestamp + b.timestamp);
+            selectQuerries(dataToSort);
         }
+    };
+    const sortByDepth = (): any => {
```

Comment:
Can `sortByDepth`, `sortByComplexity` and `sortByTime` be refactored into a single function that excepts the sort value as a parameter. This would avoid repeating what appears to be identical logic in each function.

Reply:
Maybe at some point, but right now I would leave it as it is, just until I am in charge of figuring out the chart stuff.